src/uTensor/core/types.cpp

```diff
@@ -82,24 +82,22 @@ uint16_t& TensorShape::operator[](int i) {
   return _shape[i];
 }  // Maybe handle update case
 void TensorShape::update_dims() {
+  // implicit assuming the last positive dim is the num_dims
   for (int i = 0; i < 4; i++) {
```

Comment:
@mbartling
Just as reminder, I fixed it because I think original impl is incorrect.
Let's suppose original `_num_dims` is 2 and the shape is `(2, 3, 0, 0)`.
We update the shape to `(2, 3, 1, 0)` and the updated `_num_dims` should be `3`.
However, the for loop here will return `4`.

Reply:
yeah I think you are right.